iroh-sync/src/sync.rs

```diff
@@ -456,11 +456,20 @@ impl Replica {
         let entry = Entry::new(id.clone(), record);
         let signed_entry = entry.sign(&inner.namespace, author);
         inner.peer.put(id, signed_entry.clone());
-        for cb in &inner.on_insert {
+        drop(inner);
+        let on_insert = self.on_insert.read();
+        for cb in &*on_insert {
             cb(InsertOrigin::Local, signed_entry.clone());
         }
     }
 
+    /// Hashes the given data and inserts it.
```

Comment:
Maybe make clear in the comment that the data is not stored/inserted anyhwere, only its hash. Also, maybe return the hash.

Reply:
done